zip-0226.rst

```diff
@@ -1,8 +1,307 @@
+
+
+:math:`\mathsf{assetBurn = [(v^{AssetId}, AssetId)}| \forall \mathsf{AssetId}  \textit{ s.t.}\mathsf{v^{AssetId}\neq 0}]`
+
+The value balances for each asset identifier in `assetBurn` represents the amount of that asset identifier that is being burnt. In the case of ZEC, the value balance represents either the transaction fee, or the amount of ZEC changing anonymity pools (to Sapling or Transparent).
+
+Finally, the validator needs to verify the Balance and Binding Signature by adding the value balances for all assets, as committed using their respective `AssetId` as the value base point of the Pedersen Commitment. This is done as follows
+
+:math:`\mathsf{bvk = (\sum cv_i^{net})}  - \mathsf{ ValueCommit_0^{Orchard}(v^{balanceOrchard})} - \sum_{\forall \mathsf{AssetId}\textit{ s.t. }\mathsf{v^{AssetId}\neq 0}} \mathsf{ValueCommit_0^{ORchardZSA}(v^{AssetId}, AssetId) } = \sum \mathsf{rcv_{i,j}^{net}}\mathcal{R}^{\mathsf{Orchard}}`
+
+In the case that the balance of all the action values related to a specific asset will be zero, there will be no value added to the vector. This way, the number of assets, nor their identifiers will be revealed, except in the case that an asset is burnt.
+
+**Note:** Even if this mechanism allows having transparent ↔  shielded asset transfers in theory, the transparent protocol will not be changed with this ZIP to adapt to a multiple asset structure. This means that unless future consensus rules changes do allow it, the unshielding is not not be possible for custom assets.
+
+ZSA Transaction Structure
+=========================
+Similar to version 5 transaction format, with the following modifications to the Orchard bundle, as defined in [#protocol-transactionstructure]_:
+
++-----------------+-------------+-----------------------------------+-------------------------+
+| Bytes           | Name        | Data Type                         | Description             |
++=================+=============+===================================+=========================+
+| varies          | nAssetBurn  | compactSize                       | number of assets burnt  |
++-----------------+-------------+-----------------------------------+-------------------------+
+| 40*nAssetBurn   | vAssetBurn  | bytes[40][nAssetBurn]             | 32 bytes asset type_t,  |
+|                 |             |                                   | 8 bytes of valueBalance |
++-----------------+-------------+-----------------------------------+-------------------------+
+
+And in terms of the action size, the ZSA action size differs from the Orchard action size by 32 bytes (given by the addition of the `AssetId`). This implies that the size goes from 820 bytes in the Orchard action to 852 bytes in the ZSA action.
```

Comment:
This should be part of a new ZIP for the v6 transaction format. You could start by copying the Orchard format from:
https://zips.z.cash/zip-0225

This can be a very rough draft, we'll fix up any issues later.

Reply:
The V6 transaction ZIP draft has been created [here](https://github.com/zcash/zips/pull/687), we will help add our additions in there via reviews/comments.